src/main/java/net/syscon/elite/service/BookingService.java

```diff
@@ -666,16 +670,20 @@ public MilitaryRecords getMilitaryRecords(final Long bookingId) {
 
     @Transactional
     public void updateLivingUnit(final Long bookingId, final Long livingUnitId) {
-        final var location = locationService.getLocation(livingUnitId);
         var offenderBooking = offenderBookingRepository.findById(bookingId)
-                .orElseThrow(new EntityNotFoundException(format("Offender booking for booking id %d not found", bookingId)));
-
-        if (!offenderBooking.getLocation().getId().equals(location.getAgencyId())) {
-            throw new IllegalArgumentException(format("Move to living unit in prison %s invalid for offender in prison %s", location.getAgencyId(), offenderBooking.getLocation().getId()));
-        }
+                .orElseThrow(EntityNotFoundException.withMessage(format("Offender booking for booking id %d not found", bookingId)));
+        final var location = agencyInternalLocationRepository.findById(livingUnitId)
```

Comment:
A general question around booking ID's.  Should we checking the booking is active?